server.go

```diff
@@ -88,45 +88,64 @@ func (s *Server) handleConn(conn net.Conn) {
 				}
 				ticker := time.NewTicker(timeout)
 
-				select {
-				case <-s.ctx.Done():
-					ew = writer.WriteError("CONS Server shutdown")
-				case msg := <-cons.Out():
-					ew = writer.WriteObjects(msgToRedis(msg)...)
-				case <-ticker.C:
-					ew = writer.WriteBulk(nil)
+			ConsLoop:
+				for {
+					select {
+					case <-s.ctx.Done():
+						writeErr = writer.WriteError("CONS Server shutdown")
+						break ConsLoop
+					case <-ticker.C:
+						writeErr = writer.WriteBulk(nil)
+						break ConsLoop
+					default:
```

Comment:
Why don't we push that into a `Producer` type function? It can get bigger if we enable the re-balance messages (so we can trigger commits).